contracts/savings/peripheral/Unwrapper.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { ImmutableModule } from "../../shared/ImmutableModule.sol";
+
+import { ISavingsContractV3 } from "../../interfaces/ISavingsContract.sol";
+import { IUnwrapper } from "../../interfaces/IUnwrapper.sol";
+import { IMasset } from "../../interfaces/IMasset.sol";
+import { IFeederPool } from "../../interfaces/IFeederPool.sol";
+import { IBoostedVaultWithLockup } from "../../interfaces/IBoostedVaultWithLockup.sol";
+
+contract Unwrapper is IUnwrapper, ImmutableModule {
+    using SafeERC20 for IERC20;
+
+    constructor(address _nexus) ImmutableModule(_nexus) {}
+
+    /**
+     * @dev Estimate output
+     * @param _isBassetOut    masset redemption or fpool swap
+     * @param _router         masset or feederpool
+     * @param _input          input token address
+     * @param _output         output token address
+     * @param _amount         amount
+     * @return output         Units of credits burned from sender
+     */
+    function getUnwrapOutput(
+        bool _isBassetOut,
+        address _router,
+        address _input,
+        address _output,
+        uint256 _amount
+    ) external view override returns (uint256 output) {
+        if (_isBassetOut) {
+            output = IMasset(_router).getRedeemOutput(_output, _amount);
+        } else {
+            output = IFeederPool(_router).getSwapOutput(_input, _output, _amount);
+        }
+    }
+
+    /**
+     * @dev Unwrap and send
+     * @param _isBassetOut    masset redemption or fpool swap
+     * @param _router         masset or feederpool
+     * @param _input          input token address
+     * @param _output         output token address
+     * @param _amount         amount
+     * @param _minAmountOut   min amount
+     * @param _beneficiary    beneficiary
+     * @return outputQuantity Units of credits burned from sender
+     */
+    function unwrapAndSend(
+        bool _isBassetOut,
```

Comment:
Only thing i see in this fn is that the integrator requires some knowledge about the mStable system to know whether this is true or false. Ideally would just pass the output addr

Reply:
Might be hard to do in an immutable and gas efficient way

Reply:
Maybe we can just have a view fn here that takes an mAsset addr (or imAsset), an output addr, and then returns a bool for `isBassetOut`. At least integration contracts can look this up off chain before sending the tx if they want